pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -81,8 +81,13 @@ func (s *sqlEntityServer) Init() error {
 
 	// set up the broadcaster
 	s.broadcaster, err = NewBroadcaster(context.Background(), func(stream chan *entity.Entity) error {
-		// start the poller
-		go s.poller(stream)
+		// subscribe to new events
+		if engine.DriverName() == "postgres" {
+			go s.pgWatcher(stream)
+		} else {
+			// start the poller
+			go s.poller(stream)
+		}
```

Comment:
Currently we don't have any way of signalling back whether the watcher was able to start successfully, or propagating a context to handle graceful shutdown. This was the next area I was going to look into, especially since there are several "postgres-compatible" databases that don't support notify/listen where we'd want to fall back to polling.

Reply:
I thought about this too. I was thinking about doing feature detection, like:

1. Generate a random string, could be a UUID
2. Use that string as a channel name and LISTEN to it
3. NOTIFY to that channel
4. Ping the server, say, at most 3 times and expect to see the notification back.
5. UNLISTEN from that channel to cleanup

Reply:
Might be simpler to just make this a config setting instead of trying to determine when to fall back?